src/fft.rg

```diff
@@ -279,48 +588,148 @@ function fft.generate_fft_interface(itype, dtype)
     end
   end
 
+  -- EXECUTE PLAN FUNCTIONS
+
+  -- Executes the FFT plan - GPU version. Called by execute_plan if necessary.
+  local execute_plan_gpu
+  if gpu_available then
+    __demand(__cuda, __leaf)
+    task execute_plan_gpu(input : region(ispace(itype), dtype_in),
+                          output : region(ispace(itype), dtype_out),
+                          plan : region(ispace(int1d), iface.plan),
+                          address_space : c.legion_address_space_t)
+    where reads writes (input, output, plan) do
+      var p = iface.get_plan(plan, true)
+      var proc = get_executing_processor(__runtime())
+      regentlib.assert(c.legion_processor_kind(proc) == c.TOC_PROC, "execute_plan_gpu must be executed on a GPU processor")
+      regentlib.assert(c.legion_processor_address_space(proc) == address_space, "execute_plan_gpu must be executed on a processor in the same address space")
+
+      var input_base = get_base_in(rect_in_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0]).base
+      var output_base = get_base_out(rect_out_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0]).base
+
+      cufft_assert(cufft_execute_function(p.cufft_p, [&cufft_execute_from_type](input_base), [&cufft_execute_to_type](output_base), cufft_execute_extra_args))
+    end
+  end
+
+  --- Execute plan. This is the __inline version of the task should the user wish to use that.
+  -- @param input Input to execute plan on.
+  -- @param output Output of the executed plan.
+  -- @param plan Plan object used to execute plan.
+  -- @note For GPU, it calls cufftExecZ2Z.
+  -- @note For CPU, it calls fftw_execute_dft.
   __demand(__inline)
-  task iface.execute_plan(input : region(ispace(itype), dtype),
-                          output : region(ispace(itype), dtype),
+  task iface.execute_plan(input : region(ispace(itype), dtype_in),
+                          output : region(ispace(itype), dtype_out),
                           plan : region(ispace(int1d), iface.plan))
-  where reads(input, plan), writes(output) do
+  where reads writes (input, output, plan) do
```

Comment:
Please double check that these privileges are correct. Are we writing `input`? `plan`?

Also, cosmetic but normally no space after the privilege and before `(`. So `reads(...)` or `reads writes(...)` but not `reads (...)` or `reads writes (...)`.

Reply:
Changed here: https://github.com/arjunkunna/regent-fft/commit/ba08bf54876149a276a03d839b6e85f4f7e5a7e8